File: dashboard-front/app/scripts/service/feedService.js
'use strict'
angular
  .module("dashboardFrontApp")
  .service('feedService', ['$http', 'API', 'DATE', function ($http, API, DATE) {

    this.get = get;
    this.save = save;
    this.remove = remove;
    this.getAll = getAll;

    function get(uuid) {
      return $http.get(API.BASE_URL + "/feed/" + uuid).then(function(response) {
        return response;
      }, function(error) {
        return error;
      });
    }

    function save(feed) {
      var p;
      if (feed.uuid) {
        // If update, use PUT method
        p = $http.put(API.BASE_URL + "/feed", feed);
      } else {
        // If new, use POST method
        p = $http.post(API.BASE_URL + "/feed", feed);
      }
      return p.then(function(response) {
        return response;
      }, function(error) {
        return error;
      });
    }

    function remove(uuid) {
      return $http.delete(API.BASE_URL + "/feed/" + uuid)
        .then(function(response) {
          return response;
        }, function (error){
            return error;
        });
    }

    function getAll() {
      return $http.get(API.BASE_URL + "/feed")
      .then(function(response) {
          return response;
      }, function (error){
        return error;
      });
    }

  }]);

File: dashboard-back/src/main/java/io/resourcepool/dashboard/service/FeedService.java
package io.resourcepool.dashboard.service;

import io.resourcepool.dashboard.model.metadata.Feed;

import java.util.List;

public interface FeedService {

  List<Feed> getAll();

  /**
   * Create a feed into DB.
   *
   * @param feed the feed
   */
  void save(Feed feed);

  /**
   * Update a feed into DB.
   *
   * @param feed the feed to update
   * @return the new Feed or null if failed
   */
  void update(Feed feed);

  /**
   * Delete feed from DB.
   *
   * @param uuid The feed uuid
   */
  void delete(String uuid);

  /**
   * Get feed from DB.
   *
   * @param uuid the feed uuid
   * @return the new feed or null if none
   */
  Feed get(String uuid);

  /**
   * Replace old tag by new bundle tag inside all relevant feeds.
   * @param oldTag the old tag
   * @param newTag the new tag
   * @return
   */
  void replaceTag(String oldTag, String newTag);

  /**
   * Remove all references to a bundle tag.
   * @param tag the bundle tag
   */
  void deleteBundleTag(String tag);
}
